pyproject.toml

```diff
@@ -1,16 +1,139 @@
 [build-system]
-requires = [
-    # The pin is a workaround for editable installs breaking IDEs.
-    # See this mypy issue: https://github.com/python/mypy/issues/13392#issuecomment-1213527559
-    # and our note: https://zapatacomputing.atlassian.net/wiki/spaces/ORQSRUN/pages/519012482/Editable+installs
-    "setuptools<=63",
-    "wheel",
-    "setuptools_scm>=6.0",
+requires = ["hatchling", "hatch-vcs"]
+build-backend = "hatchling.build"
+
+[project]
+authors = [
+    { name = "Zapata Computing Inc.", email = "info@zapatacomputing.com" },
+]
+name = "orquestra-sdk"
+description = "Compose Orquestra workflows using a Python DSL"
+readme = "README.md"
+license = {file = "LICENSE"}
+classifiers = [
+    "Programming Language :: Python :: 3",
+    "Operating System :: OS Independent",
+    "License :: OSI Approved :: Apache Software License",
+    "Topic :: Scientific/Engineering",
+]
+dynamic = ["version"]
+requires-python = ">= 3.8"
+
+# As a rule of thumb, the dependencies should be specified with lower bound
+# only.
+#
+# For pratical reasons, when choosing the lower bound, we pick the oldest
+# version we can easily install and test, even if there are older versions with
+# the same API.
+dependencies = [
+    # Schema definition
+    "pydantic>=1.10.8,<2",
+    # Pickling library
+    "cloudpickle==2.2.1",
+    # We should keep `dill` to make sure old workflows can be unpickled.
+    "dill==0.3.6",
+    # Enhances wrapping of user's functions
+    "wrapt",
+    # Configuration file locking
+    "filelock>=3.3.2",
+    # Parse version strings
+    "packaging>=21",
+    # For different import types in SDK
+    "GitPython",
+    "pip-api==0.0.30",
+    # Python Standard Library Backports
+    "importlib_metadata~=4.0",
```

Comment:
**thought**: do you think we should also (in another PR) change all `~=` into `>=`?

Reply:
I think we should only do this if/when we have "nightly" runs of the test suite.

Otherwise, an update may be released for a dependent package with a breaking change, which bumps semver, and we miss it.

Perhaps we should do this already 😬

Reply:
Nightly runs: great idea! I think it's just a matter of combining [cron triggers](https://jasonet.co/posts/scheduled-actions/) with disabling pip cache (or another approach that ensures we get the latest and greatest dependencies)

Reply:
ORQSDK-1022